joblib/memory.py

```diff
@@ -131,10 +131,20 @@ def _load_output(output_dir, func_name, timestamp=None, metadata=None,
                     t, str(signature), output_dir))
 
     filename = os.path.join(output_dir, 'output.pkl')
-    if not os.path.isfile(filename):
+
+    n_attempts = 3
+    sleep_time = 0.1
+
+    for _ in range(n_attempts):
```

Comment:
Not sure if there is a better solution than this. This is necessary in the case where the second process sees the output folder so tries to load the result but fails because the output.pkl has not been created yet.

I'll try to see if creating the temporary pickle in a separate folder e.g. tempfile.NamedTemporaryFile makes this issue less likely to happen. I believe this race condition will still exists though. Any better solution or tweaks on this solution well appreciated.